mesonbuild/scripts/depfixer.py

```diff
@@ -432,11 +432,12 @@ def fix_jar(fname):
 
 def fix_rpath(fname, new_rpath, final_path, install_name_mappings, verbose=True):
     global INSTALL_NAME_TOOL
-    # Static libraries never have rpaths
-    if fname.endswith('.a'):
+    # Static libraries, import libraries, debug information, headers, etc
+    # never have rpaths
+    if fname.endswith(('.a', '.lib', '.pdb', '.h', '.hpp')):
         return
-    # DLLs and EXE never have rpaths
-    if fname.endswith('.dll') or fname.endswith('.exe'):
+    # DLLs and EXE currently do not need runtime path fixing
+    if fname.endswith(('.dll', '.exe')):
```

Comment:
nitpicking: why not adding dll/exe in the same list above?

Reply:
I guessed the original code kept it separate because they were skipped for semantically different reasons (don't have RPATHs vs [we haven't implemented it yet](https://nibblestew.blogspot.com/2019/05/emulating-rpath-on-windows-via-binary.html)), so I kept it that way.

Reply:
Hm, maybe... I think it was more because the one who write that didn't know the tuple syntax, so it would become quickly a long condition (I just learnt it now).

Reply:
But since you're mixing multiple semantics in the first condition already, I would sneark exe/dll too. But again, just a little nitpicking, feel free to ignore :-)